### PR TITLE
PWA

### PR DESCRIPTION
**Pull request by alianza**
Progressive Web Application implementation using next-pwa

Added next-pwa package
Added next-pwa config to next.config (Webpack 5 upgrade was required for offline fallback page)
Added Offline fallback page
Updated webmanifest

### CODE REVIEW

File: pages/_offline.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        <div className="flex-1 p-24 flex flex-col justify-center items-center ">
+        <span
+          className="border border-dashed border-secondary rounded-full flex items-center justify-center w-16 h-16 p-12 bg-primary text-primary">
+          <Bars className="absolute"/>
+        </span>
+          <h2 className="pt-6 text-2xl font-bold tracking-wide text-center">
+            No internet connection...
+          </h2>
+          <p className="text-accents-6 px-10 text-center pt-2">
+            Please reconnect to the network to access the website. Click <a className="underline" href={'/'} onClick={() => location.reload()}>here</a> to try again!
```
Comment (egonm12) [inline pages/_offline.tsx:22]: This way you will reload the page that you are currently on
```suggestion
            Please reconnect to the network to access the website. Click <a className="underline" href='.' onClick={() => location.reload()}>here</a> to try again!
```

**Review by egonm12 — COMMENTED** [review]
Looks good overall! I wonder if it would be better to show the offline page as well when navigating between pages and you go offline 🤔 . Maybe something to discuss with @pascalw?

**Comment by pascalw** [comment]
When exactly do you get the offline page now? I went offline, but didn't get it.
In general I would prefer a small (dismissable) banner/notification that notifies the user their offline, instead of a full offline page (that kinda defeats the purpose).

**Comment by pascalw** [comment]
Btw offline support in general is actually quite tricky. The experience is pretty bad now, because many things can fail and there doesn't seem to be a lot of error handling. For example if you try to add something to the cart when offline, it just fails without any user feedback.

**Comment by alianza** [comment]
> When exactly do you get the offline page now? I went offline, but didn't get it.
> In general I would prefer a small (dismissable) banner/notification that notifies the user their offline, instead of a full offline page (that kinda defeats the purpose).

When you go offline and navigate to a non-cached page the offline fallback page should appear. I don't what what is possible in terms of a small dismissable banner/notification about the user being offline. Maybe this could be a separate ticket as this PR does achieve the current goals.

**Comment by alianza** [comment]
> Btw offline support in general is actually quite tricky. The experience is pretty bad now, because many things can fail and there doesn't seem to be a lot of error handling. For example if you try to add something to the cart when offline, it just fails without any user feedback.

Yes I noticed that. There isn't proper error handling for that in place. Maybe merging the original Next.js Commerce repo with all the updates since will bring some improvements with it.

**Comment by egonm12** [comment]
> > When exactly do you get the offline page now? I went offline, but didn't get it.
> > In general I would prefer a small (dismissable) banner/notification that notifies the user their offline, instead of a full offline page (that kinda defeats the purpose).
> 
> When you go offline and navigate to a non-cached page the offline fallback page should appear. I don't what what is possible in terms of a small dismissable banner/notification about the user being offline. Maybe this could be a separate ticket as this PR does achieve the current goals.

I am still able to navigate to the products page when caching is disabled. The offline page only appears when clicking on the footer links and when reloading in the products page for example.

**Comment by pascalw** [comment]
Well let's not sweat the details now, for demo/POC purposes this is fine :-)

**Review by pascalw — APPROVED** [review]

**Review by egonm12 — APPROVED** [review]

**Pull request merged**